[Themes] New example for custom themes

romoore opened the pull request:
Supports Primary and Secondary colors using MDCBasicColorScheme.

![simulator screen shot - iphone 4s - 2017-08-25 at 17 35 47](https://user-images.githubusercontent.com/1753199/29733584-ea7b8d42-89bb-11e7-881a-52e84a248caf.png)

Commit:
[Themes] New example for custom themes

Supports Primary and Secondary colors using MDCBasicColorScheme.

Commit:
Formatting

ajsecord reviewed (changes requested):
Let's use MDCTextField for this.

Commit:
Fixing widths, dropping labels

Commit:
Formatting

willlarche reviewed (changes requested):
@romoore , The issue with the text fields is the space they require. This version of text fields requires 63.5 points for non-floating and 84.5 points for floating heights. 

The simplest solution is to increase the square's width and height to 64 and keep it non-floating.

Though I would implement a tap gesture recognizer and / or textFieldShouldReturn to resign first responder. It's a little maddening at the moment if you want to get the keyboard off the screen.

romoore commented:
Awesome, @willlarche,  thanks for investigating!

Commit:
Updating constraints

Commit:
Organizing example a bit

romoore commented:
Added minimum height constraint for both text fields (64pt, keeping non-floating). Added gesture recognizer to resign first responder on both text fields.

Commit:
Refactoring the string parsing code

Commit:
Fixing return value

willlarche commented on components/Themes/examples/resources/ThemerCustomSchemePicker.storyboard:
```diff
@@ -0,0 +1,132 @@
+                                        <rect key="frame" x="0.0" y="24" width="160" height="64"/>
+                                        <constraints>
+                                            <constraint firstAttribute="width" priority="750" constant="160" id="C1a-8M-aWi"/>
+                                            <constraint firstAttribute="height" relation="greaterThanOrEqual" constant="64" id="VQR-Rf-sMK"/>
```
Much better. Best would be no height constraint at all.

Commit:
Tweaking height constraints

Commit:
Catalog by convention fix
